### PR TITLE
only check nonces of transactions who's from address match the txMeta

### LINKED ISSUE #2810: transaction says failed, but succeeded
The latest transaction in my MetaMask history says it failed (`NonceTakenErr`), but when I click on the transaction and view it in etherscan, I can see that it succeeded just fine.

In case it helps: This transaction was generated by MEW to call a function in a contract in the Ropsten test network.

Screenshot:

screen shot 2017-12-25 at 7 22 39 pm

### PR DESCRIPTION
**Pull request by frankiebee**
fixes #2810 

the issue seemed to be that we were checking against all confirmed transactions instead of just the ones for that specific address

### CODE REVIEW

**Comment by Zanibas** [comment]
Is it possible for us to write tests for this? I'd like for us to get into the habit of writing tests for our PRs.

File: test/unit/pending-tx-test.js [hunk 1]
```diff
@@ -338,5 +338,64 @@ describe('PendingTransactionTracker', function () {
...
+        },
+        rawTx: '0xf86c808504a817c800827b0d940c62bb85faa3311a998d3aba8098c1235c564966880de0b6b3a7640000802aa08ff665feb887a25d4099e40e11f0fef93ee9608f404bd3f853dd9e84ed3317a6a02ec9d3d1d6e176d4d2593dd760e74ccac753e6a0ea0d00cc9789d0d7ff1f471d',
+      }]
+      pendingTxTracker.getCompletedTransactions = (address) => {
+        if (!address) throw new Error('unless behavior has changed #_checkIfNonceIsTaken needs a filtered list of transactions to see if the nonce is taken')
+        return confirmedTxList
+      }
+    })
+
+    it('should return false', function (done) {
```
Comment (Zanibas) [inline test/unit/pending-tx-test.js]: Can we clarify when these should return false?

File: test/unit/pending-tx-test.js [hunk 2]
```diff
@@ -338,5 +338,64 @@ describe('PendingTransactionTracker', function () {
...
+        },
+      })
+      .then((taken) => {
+        assert.ok(!taken)
+        done()
+      })
+      .catch(done)
+    })
+
+    it('should return true', function (done) {
```
Comment (Zanibas) [inline test/unit/pending-tx-test.js]: Same for here when they return true. Sorry for being pedantic, but future generations will thank us.

**Review by 2-am-zzz — CHANGES_REQUESTED** [review]
Other than the small descript changes, this should be good to go.

**Review by 2-am-zzz — APPROVED** [review]

**Pull request merged**